### DISCUSSION TITLE
有没有推荐的并行编程学习教程或者书籍？

### DISCUSSION THREAD

**Discussion by liu-mengyang** [mlc-ai/mlc-zh#111] [Q&A] [answered]
学习最新一节课，终于涉及到了并行硬件上的MLC优化，但感觉很多优化原理和老师说到的概念需要结合一些并行编程基础进行食用，我自己受到的CS课程教学中没有系统地学到相关知识，请问能不能推荐一些tutorial、book以及lecture呢？

**Comment by liu-mengyang** [comment, marked as answer]
我注意到 [mlc-zh/discussions/108](https://github.com/mlc-ai/mlc-zh/discussions/108) 提到了类似问题，但是我希望学到的内容不仅仅是针对GPU或者CUDA编程的，而是更高一级的并行编程技术内容 🙏

**Reply by Hzfengsy** [reply]
不太清楚你所说的“更高一级的并行编程技术”具体指怎么样的内容。是指更更多的优化细节还是针对其他硬件的编程技术？

**Reply by liu-mengyang** [reply]
我所求的就是想要比较全面的关于并行计算的资源推荐，它涵盖了一些并行计算的常见概念介绍和并行编程进行优化时常常需要使用的技术，比如这节课老师说的利用层次存储模型减少访存次数，还有一些并行编程思想，比如课上老师提到的reduce和spatial allocation（这个我可能记得不是太清楚）。请问有推荐的相关资源或者教程吗？

**Reply by Hzfengsy** [reply]
考虑去搜索一下国外大学的相关课程，比如 https://gfxcourses.stanford.edu/cs149/fall21/lecture/

**Reply by liu-mengyang** [reply]
好嘞，谢谢！
